Added support for g2g share build

mtakal opened the pull request:
Memory sharing between guests requires a suitable device tree file
    for the host.
    When the host is booted in qemu, the appropriate device tree is
    created when the make parameter is USE_G2GSHARE=1
    e.g.: make USE_G2GSHARE=1 run

psittacus-erithacus committed:
pkvm-debug-tools: Update to the new version

Pkvm debug tool has some improvements

Signed-off-by: Martti Takala <martti.takala@unikie.com>

vrosendahl commented on platform/virt/Makefile:
```diff
@@ -13,6 +13,8 @@ PVMFW_ADDRESS := 0x40001000
 
 INITRAMFS := $(HOST_IMAGE_DIR)/initramfs.gz
 INITRAMFS_ADDR := 0x7f000000
+#G2G_SHARE_SIZE := 0x400000
```
I dislike commented out code in general. At minimum there should be some kind of explanation of in what kind of situations the larger size might be useful, i.e. "if you startx guests with option X, you might want a larger size"

psittacus-erithacus committed:
build files: Added support for g2g share build

Memory sharing between guests requires a suitable device tree file
for the host.
When the host is booted in qemu, the appropriate device tree is
created when the make parameter is USE_G2GSHARE=1
e.g.: make USE_G2GSHARE=1 run

Removed some old stuff from platform makefile

Signed-off-by: Martti Takala <martti.takala@unikie.com>

vrosendahl commented on Makefile:
```diff
@@ -1,6 +1,7 @@
 include core/vars.mk
 
 DIRS := tools qemu-user host-kernel ubuntu-template target-crosvm hostimage guest-kernel guestimage
+TMPDIR := $(shell mktemp -d)
```
This will create a new directory in /tmp when the Makefile is being parsed and it will never be deleted. It will also be created even if I execute an unrelated target like "make guest-kernel-clean".

vrosendahl reviewed (changes requested):
Maybe I am missing something but this is now leaking temporary directories that are never being removed. They are also being created when they are not needed. See comment below.
